lib/store.ml

```diff
@@ -21,13 +21,15 @@ end
 module type S =
 sig
   type t
+  type 'a merge_field = 'a option -> 'a option -> 'a option
```

Comment:
Shouldn't we use a dedicated datatype? So that we ensure that `None | None` is always merged to `None`.